### DISCUSSION TITLE
Huawai Sound X

### DISCUSSION THREAD

**Discussion by bloop16** [philippe44/AirConnect#490] [Show and tell]
Hello,

I think i made my Huawai Sound X running with Airupnp.
Version 1.2.2 on Debian 11.7
I used the following config:

```
<?xml version="1.0"?>
<airupnp>
<common>
<protocolInfo>
<pcm>http-get:*:audio/L16;rate=44100;channels=2:DLNA.ORG_PN=LPCM;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=0d500000000000000000000000000000</pcm>
<wav>http-get:*:audio/wav:DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=0d500000000000000000000000000000</wav>
<flac>http-get:*:audio/flac:DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=0d500000000000000000000000000000</flac>
<mp3>http-get:*:audio/mpeg:DLNA.ORG_PN=MP3;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=0d500000000000000000000000000000</mp3>
</protocolInfo>
<enabled>1</enabled>
<max_volume>100</max_volume>
<http_length>0</http_length>
<upnp_max>1</upnp_max>
<codec>mp3:320</codec>
<metadata>0</metadata>
<flush>0</flush>
<artwork>0</artwork>
<latency>rtc:http:500</latency>
<drift>1</drift>
</common>
<main_log>info</main_log>
<upnp_log>info</upnp_log>
<util_log>warn</util_log>
<raop_log>info</raop_log>
<log_limit>-1</log_limit>
<max_players>32</max_players>
<binding>?</binding>
<ports>0:0</ports>
<device>
<udn>uuid:e60387b3-e2b5-d9b9-0c57-915492efb6d9</udn>
<name>osmc+</name>
<mac>bb:bb:01:68:71:26</mac>
<enabled>0</enabled>
</device>
</airupnp>
```

and run the service with

```./airupnp-linux-x86_64 -Z -x /var/lib/airconnect/config.xml```

A view last problems to solve:

- i always need to start playing twice or more times
- delay between handheld and sound device is about 10 seconds

Anyone else with the same issue?

**Comment by philippe44** [comment]
What did you change to make it work?

**Reply by bloop16** [reply]
Made an update and set HTTP length to 0 and flush to 0.

Thing is that it takes about 10 to 15sec until music starts playing and for example Spotify stops again. You then have to press start again. Once Sound X is giving out music everything works fine. I think the problem is the buffering time in the beginning.
A noter interesting thing is, that volume change get recognized nearly direct.
